### PR TITLE
Task/basic transpilation gim 117

### PR DESCRIPTION
**Pull request by sbilge**
Transpiling Excel workbook to a JSON object. Closes GIM-117. Processing spreadsheet version will be in a separate PR. 
- Config yaml
- Config parser
- Excel to json converter

### CODE REVIEW

File: ghga_transpiler/cli.py [hunk 1]
```diff
@@ -20,20 +20,36 @@
...
+                workbook[sheet.sheet_name],
+                sheet.settings.start_row,
+                workbook[sheet.sheet_name].max_row,
+                sheet.settings.start_column,
+                sheet.settings.end_column,
+            )
+        except KeyError as exc:
+            raise MissingWorkbookContent(
+                f"Workbook does not contain {sheet.sheet_name} worksheet."
+            ) from exc
```
Comment (lkuchenb) [inline ghga_transpiler/cli.py]:
```suggestion
            )
```
You are catching an error that is based on invalid user input. We know the cause, providing the user with the exception trace is not helpful. In fact, this exception should be caught in the cli command and transformed into a regular CLI error message.

File: ghga_transpiler/cli.py [hunk 2]
```diff
@@ -20,20 +20,36 @@
...
 
-    return filename
+        converted_workbook[sheet.sheet_name] = convert_rows(get_header(rows), rows[1:])
```
Comment (lkuchenb) [inline ghga_transpiler/cli.py]: Shouldn't this hardcoded `1` be the start row?

Comment (lkuchenb) [inline ghga_transpiler/cli.py]: `sheet.sheet_name` should be the attribute name from the config, right?

File: ghga_transpiler/core/__init__.py [hunk 3]
```diff
@@ -13,7 +13,5 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-This sub-package contains the main business functionality of this service.
-It should not contain any service API-related code.
-"""
+"""Module to load core functions for workbook processing"""
+from .core import convert_rows, get_header, get_worksheet_rows, read_workbook  # noqa
```
Comment (lkuchenb) [inline ghga_transpiler/core/__init__.py]:
```suggestion
from .core import (  # noqa: F401
    convert_rows,
    get_header,
    get_worksheet_rows,
    read_workbook,
)
```
Be specific which error you want the linter to ignore, in this case [Module imported but unused](https://www.flake8rules.com/rules/F401.html).

File: ghga_transpiler/core/core.py [hunk 4]
```diff
@@ -14,4 +14,41 @@
...
+    """
+    return load_workbook(filename)
+
+
+def get_worksheet_rows(
+    worksheet,
+    min_row: Union[int, None],
+    max_row: int,
+    min_col: Union[int, None],
+    max_col: Union[int, None],
```
Comment (lkuchenb) [inline ghga_transpiler/core/core.py]:
```suggestion
    min_row: Optional[int],
    max_row: int,
    min_col: Optional[int],
    max_col: Optional[int],
```
But when is this actually ever `None`?

File: ghga_transpiler/core/core.py [hunk 5]
```diff
@@ -14,4 +14,41 @@
...
+
+
+def convert_rows(header: list, rows: list) -> list:
+    """Function to return list of dictionaries, rows as values and column names as keys"""
+    return [dict(zip(header, row)) for row in rows]
+
+
+def get_header(worksheet_rows: list[list]) -> list[str]:
+    """Function to return a list column names of a worksheet"""
+    return worksheet_rows[0]
```
Comment (lkuchenb) [inline ghga_transpiler/core/core.py]: We should actually also have the header row configurable, I think in an early config schema we also had it configurable.

File: ghga_transpiler/cli.py [hunk 6]
```diff
@@ -50,7 +66,7 @@ def cli_main(
         False, "--force", "-f", help="Override output file if it exists."
     ),
 ):
-    """Function to convert excel spread sheet to JSON"""
+    """Function to get options and channel those to the convert workbook functionality"""
 
     if output_file is None:
         print(convert_workbook(spread_sheet))
```
Comment (lkuchenb) [inline ghga_transpiler/cli.py]: The output is not converted to JSON

**Review by lkuchenb — CHANGES_REQUESTED** [review]
The `config.configs` module survived the merge, should be deleted.

File: ghga_transpiler/cli.py (hunk 1 above)
Comment (sbilge) [reply ghga_transpiler/cli.py]: Pylint insists on[ `as exc` `from exc`](https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/raise-missing-from.html)

File: ghga_transpiler/process_workbook.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+
+def read_workbook(filename: str) -> Workbook:
+    """
+    Function to read-in a workbook (aka spreadsheet)
+    """
+    return load_workbook(filename)
+
+
+def get_version(workbook):
```
Comment (lkuchenb) [inline ghga_transpiler/process_workbook.py]:
```suggestion
def get_version(workbook) -> str:
```

File: ghga_transpiler/process_workbook.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    """Function to get workbook version from the worksheet _properties"""
+    if "__properties" in workbook.sheetnames:
+        return workbook["__properties"].cell(1, 1).value
```
Comment (lkuchenb) [inline ghga_transpiler/process_workbook.py]:
```suggestion
        return str(workbook["__properties"].cell(1, 1).value)
```

File: ghga_transpiler/process_workbook.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+def params(filename: Path) -> Tuple[Workbook, Config]:
+    """Helper function to create workbook, config and channel them into convert_workbook"""
+    workbook = read_workbook(str(filename))
+    config = load_config(get_version(workbook))
+    return workbook, config
+
+
+def convert_workbook(workbook: Workbook, config: Config):
+    """Function to convert an input spreadsheet into JSON"""
```
Comment (lkuchenb) [inline ghga_transpiler/process_workbook.py]:
```suggestion
def convert_workbook(workbook: Workbook, config: Config) -> dict:
    """Function to convert an input spreadsheet to a dictionary"""
```

File: ghga_transpiler/process_workbook.py [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+"""This module contains functionalities for processing excel sheets into json object."""
+from pathlib import Path
+from typing import Tuple, Union
+
+from openpyxl import Workbook, load_workbook
+
+from .config import Config, load_config
+from .config.exceptions import MissingWorkbookContent
+
```
Comment (lkuchenb) [inline ghga_transpiler/process_workbook.py]: Multiline comment

**Review by lkuchenb — DISMISSED** [review]
Good to go now 👍

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**